Use parametrized tests instead of dynamic test functions

orf opened the pull request:
[Parametrized tests](https://docs.pytest.org/en/latest/example/parametrize.html) have the same effect, but with a lot less boilerplate.

windelbouwman commented on tests/test_snippets.py:
```diff
@@ -39,27 +39,37 @@ def pushd(path):
+def test_rustpython(test_type, filename):
+    env = os.environ.copy()
+    log_level = 'info' if test_type == _TestType.benchmark else 'trace'
+    env['RUST_LOG'] = '{},cargo=error,jobserver=error'.format(log_level)
```
Could you de-duplicate this code by storing it into a function?

windelbouwman commented:
I wonder wether this is an improvement. Now we can only run the tests with pytest (which most people do anyway, I agree, but still). Also, the code size was not reduced that much. Are there more benefits I am missing?

I will merge this anyway if you create a helper function with the duplicate code logic.

orf commented:
I've added a utility method. IMO it makes it a bit more readable than dynamically creating the tests, and it feels more pytest'y. On the readability front, I switched to using `pathlib.Path` because it feels much more readable than a lot of nested `os.path.*` methods. I can remove that commit if you disagree.

windelbouwman commented:
I agree about the pathlib! What is the problem with the CI at this point?

windelbouwman commented:
Closing down this pull request. Please re-open if required.
